Any program built before January 4 2022 used a flatpak builder version we could not control, and are therefore not reproduced in the right environment. Here we look how many were affected.

In [1]:
%matplotlib notebook

import numpy as np
import matplotlib.pyplot as plt
from matplotlib.widgets import Slider, Button
import json
import os
from helpers import extract_stats, run_flatpak_command, cmd_output_to_dict, flatpak_date_to_datetime, get_additional_deps, GitNotFoundException, find_build_manifest
import subprocess
import shutil
from git.repo import Repo
import datetime as dt
from datetime import datetime, timezone
import yaml
from functools import reduce
import json5

In [2]:
results = extract_stats('../results/rebuild_half_flathub/')

In [3]:
def used_wrong_fb(name):
    cmd = ['flatpak', 'remote-info', 'flathub', name]
    try:
        output = run_flatpak_command(cmd, 'system', capture_output=True)
        commit = cmd_output_to_dict(output)
        commit_date = flatpak_date_to_datetime(commit["Date"])
    
        return commit_date < datetime(2022, 1, 4, tzinfo=timezone.utc)
    except:
        return False

In [4]:
wrong = [stat for stat in results if stat['build_sucess'] and used_wrong_fb(stat['name'])]

In [5]:
print(len(wrong))

180


In [6]:
wrong_and_not_repro = [stat for stat in wrong if not stat['is_reproducible']]

In [7]:
print(len(wrong_and_not_repro))

132


As of today (8 June 2022) 180 programs from the rebuilding experiment were not using the right flatpak-builder version, and (concequently maybe) 132 were not reproducible. Those numbers are a lower-bound since the original rebuild process took place in May (I think I cannot remember), and some programs were updated since then.